scripts/modify-source.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+srcFilePath=$1
+permission_type=$2
+
+# java does not support string constants beyond 64K size
+# this script processes solc generated java files to split up BYTECODE into multiple lines of smaller data size.
+# this allows java compilation to pass for solc contracts with bytecode size > 64k
+echo "process java source files from $srcFilePath ...permission_type=$permission_type"
+gawk --version
+
+cd $srcFilePath
+for file in `ls -1 ./*.java` ; do
+    if [ -f $file ]; then
+        echo "Processing $file"
+        mv "$file" "$file.bk"
+        gawk '{ if( match($0, /^(.*?String BINARY = )"([^"]+)"/, m))
+            {
+                print m[1], "new StringBuilder()"
+                n = split (m[2], ary, /.{1,200}/, seps);
+                for (i=1; i<=n; i++)
+                   printf ".append(\"%s\")%s", seps[i], (i<n ? "\n" : ".toString();\n")
+           } else {
+                print $0
+           }
+        }' "$file.bk" > "$file"
+        rm "$file.bk"
+    fi
+done
```

Comment:
I would prefer to support platform agnostic. 

Can you give a try with `maven-replacer-plugin` or `spoon-maven-plugin` instead? 

Noted that the binary is also generated in a `.bin` file so the `BINARY` string can be read like: `SimpleStorage.class.getResourceAsStream("/com.quorum.gauge.sol/SimpleStorage.bin");` which makes the regex simpler

Reply:
I agree with you, but those plugins don't provide an easy way to split the binary string to multiple lines of equal size strings. Thats why I started using shellscript. 

Reading the binary from `.bin` file has its own challenges. We need to modify a oneline code to something like ` private static final String BINARY =  new String(SimpleStorage.class.getResourceAsStream("/com.quorum.gauge.sol/SimpleStorage.bin").readAllBytes());
` and this throws IOException and it can't be handled when you initialise class level variables.
And we can't import any Util packages to make this simpler because its auto generated code.

Reply:
> And we can't import any Util packages to make this simpler because its auto generated code.

Fully-qualified path like `org.apache.commons.io.IOUtils.toString(...)` or something similar may not need import

Reply:
I have done the changes with `maven-antrun-plugin`'s `replace`. it works fine.